decidim-generators/lib/decidim/generators/test/generator_examples.rb

```diff
@@ -539,28 +721,349 @@
+      .and match(/gem ["']+google-cloud-storage["']+/)
+
+    services.each do |service|
+      current = rails_value("Rails.application.config.active_storage.service", test_app, storage_envs.merge({ "STORAGE_PROVIDER" => service }))
+      expect(current).to eq(service), "Rails storage service (#{current}) expected to match provider (#{service})"
+    end
+  end
+end
+
+shared_examples_for "an application with storage and queue gems" do
+  let(:queue_envs_off) do
+    {
+      "RAILS_ENV" => "production"
+    }
+  end
+  let(:queue_envs_on) do
+    {
+      "RAILS_ENV" => "production",
+      "QUEUE_ADAPTER" => "sidekiq",
+      "SIDEKIQ_CONCURRENCY" => "11"
+    }
+  end
+
+  it "includes storage and queue gems in the Gemfile" do
+    expect(result[1]).to be_success, result[0]
+
+    expect(File.read("#{test_app}/Gemfile"))
+      .to match(/gem ["']+aws-sdk-s3["']+/)
+      .and match(/gem ["']+sidekiq["']+/)
+
+    current = rails_value("Rails.application.config.active_job.queue_adapter", test_app, queue_envs_off)
+    expect(current).to eq("async"), "sidekiq queue (#{current}) expected to be async"
+
+    current = rails_value("Rails.application.config.active_job.queue_adapter", test_app, queue_envs_on)
+    expect(current).to eq("sidekiq"), "sidekiq queue (#{current}) expected to be sidekiq"
+    current = rails_value("YAML.load(ERB.new(IO.read(\"config/sidekiq.yml\")).result)", test_app, queue_envs_off)
+    expect(current["concurrency"]).to eq(5), "sidekiq concurrency (#{current["concurrency"]}) expected to eq 5"
+
+    current = rails_value("YAML.load(ERB.new(IO.read(\"config/sidekiq.yml\")).result)", test_app, queue_envs_on)
+    expect(current["concurrency"]).to eq(11), "sidekiq concurrency (#{current["concurrency"]}) expected to eq 11"
+
+    queues = %w(mailers vote_reminder reminders default newsletter newsletters_opt_in conference_diplomas events translations user_report block_user metrics exports)
+    expect(current["queues"].flatten).to include(*queues), "sidekiq queues (#{current["queues"].flatten}) expected to eq containt (#{queues})"
+  end
 end
 
 def json_secrets_for(path, env)
-  JSON.parse Decidim::GemManager.new(path).capture("bin/rails runner 'puts Rails.application.secrets.to_json'", env: env, with_stderr: false)[0]
+  JSON.parse capture(path, "bin/rails runner 'puts Rails.application.secrets.to_json'", env: env)
 end
 
-def initializer_config_for(path, env)
-  JSON.parse Decidim::GemManager.new(path).capture("bin/rails runner 'puts Decidim.config.to_json'", env: env, with_stderr: false)[0]
+def initializer_config_for(path, env, mod = "Decidim")
+  JSON.parse capture(path, "bin/rails runner 'puts #{mod}.config.to_json'", env: env)
 end
 
 def rails_value(value, path, env)
-  JSON.parse Decidim::GemManager.new(path).capture("bin/rails runner 'puts #{value}.to_json'", env: env, with_stderr: false)[0]
+  JSON.parse capture(path, "bin/rails runner 'puts #{value}.to_json'", env: env)
+end
+
+def capture(path, cmd, env: {})
```

Comment:
I think we should move these helper methods inside the examples/context.

Otherwise these methods will be in the global namespace after this test file has been loaded.

It can cause conflicts some times, especially with the method name `capture`.

Reply:
good point, thanks